django_auth_adfs/token_manager.py

```diff
@@ -0,0 +1,489 @@
+
+                    backend = AdfsBaseBackend()
+                    obo_token = backend.get_obo_access_token(access_token)
+                    if obo_token:
+                        encrypted_token = self.encrypt_token(obo_token)
+                        if encrypted_token:
+                            request.session[self.OBO_ACCESS_TOKEN_KEY] = encrypted_token
+                            import jwt
+
+                            decoded_token = jwt.decode(
+                                obo_token, options={"verify_signature": False}
+                            )
+                            if "exp" in decoded_token:
+                                obo_expires_at = datetime.datetime.fromtimestamp(
+                                    decoded_token["exp"]
+                                )
+                                request.session[self.OBO_TOKEN_EXPIRES_AT_KEY] = (
+                                    obo_expires_at.isoformat()
+                                )
+                                session_modified = True
+                                logger.debug(
+                                    "Stored OBO token with expiration from token claims"
+                                )
+                except Exception as e:
+                    logger.warning(f"Error getting OBO token: {e}")
+
+            if session_modified:
+                request.session.modified = True
+                logger.debug("All tokens stored successfully")
+                return True
+
+            logger.warning("No tokens were stored")
+            return False
+
+        except Exception as e:
+            logger.warning(f"Error storing tokens in session: {e}")
+            return False
+
+    def _get_encryption_key(self):
+        """
+        Derive a Fernet encryption key from Django's SECRET_KEY.
+
+        Returns:
+            bytes: A 32-byte key suitable for Fernet encryption
+        """
+        # Use Django's SECRET_KEY to derive a suitable encryption key
+        default_salt = b"django_auth_adfs_token_encryption"
+        salt = getattr(settings, "TOKEN_ENCRYPTION_SALT", default_salt)
+
+        if isinstance(salt, str):
+            salt = salt.encode()
+
+        kdf = PBKDF2HMAC(
+            algorithm=hashes.SHA256(),
+            length=32,
+            salt=salt,
+            iterations=100000,
+        )
+        key = base64.urlsafe_b64encode(kdf.derive(django_settings.SECRET_KEY.encode()))
+        return key
+
+    def encrypt_token(self, token):
+        """
+        Encrypt a token using Django's SECRET_KEY.
+
+        Args:
+            token (str): The token to encrypt
+
+        Returns:
+            str: The encrypted token as a string or None if encryption fails
+        """
+        if not token:
+            return None
+
+        try:
+            key = self._get_encryption_key()
+            f = Fernet(key)
+            encrypted_token = f.encrypt(token.encode())
+            return encrypted_token.decode()
```

Comment:
Would we save ourselves some code (and tests) if we rely on [Django's signing API](https://docs.djangoproject.com/en/5.1/topics/signing/)?